swim/join_sender.go

```diff
@@ -373,49 +375,38 @@ func (j *joinSender) JoinGroup(nodesJoined []string) ([]string, []string) {
 	j.numTries++
 	j.node.emit(JoinTriesUpdateEvent{j.numTries})
 
-	for _, node := range group {
```

Comment:
Great change!

At this point, you know really well what `JoinGroup` is doing. Can you take some time to document the function? Especially, the input and output arguments, and how it does error reporting.